cpp/src/arrow/flight/sql/odbc/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cmake_minimum_required(VERSION 3.16)
+set(CMAKE_CXX_STANDARD 17)
+
+add_custom_target(arrow_flight_sql_odbc)
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+  add_compile_definitions(NDEBUG)
+endif()
+
+# Add Boost dependencies. Should be pre-installed (Brew on Mac).
+find_package(Boost REQUIRED)
```

Comment:
Could you remove this and update https://github.com/apache/arrow/blob/598938711a8376cbfdceaf5c77ab0fd5057e6c02/cpp/cmake_modules/ThirdpartyToolchain.cmake#L1251-L1273 instead?

BTW, do we really need Boost? For example, `boost/optional.h` can be replaced with `std::optional` in C++17.

Reply:
removed `find_package` for Boost, updated `ThirdpartyToolchain.cmake`